lib/project_types/script/layers/application/push_script.rb

```diff
@@ -9,29 +9,36 @@ def call(ctx:, force:, project:)
             script_project_repo = Infrastructure::ScriptProjectRepository.new(ctx: ctx)
             script_project = script_project_repo.get
             script_project.env = project.env
+
             task_runner = Infrastructure::Languages::TaskRunner
               .for(ctx, script_project.language, script_project.script_name)
 
-            extension_point = ExtensionPoints.get(type: script_project.extension_point_type)
-            library_name = extension_point.libraries.for(script_project.language)&.package
-            raise Infrastructure::Errors::LanguageLibraryForAPINotFoundError.new(
-              language: script_project.language,
-              api: script_project.extension_point_type
-            ) unless library_name
+            if task_runner.respond_to?(:build)
+              extension_point = ExtensionPoints.get(type: script_project.extension_point_type)
+              library_name = extension_point.libraries.for(script_project.language)&.package
+              raise Infrastructure::Errors::LanguageLibraryForAPINotFoundError.new(
+                language: script_project.language,
+                api: script_project.extension_point_type
+              ) unless library_name
+
+              library = {
+                language: script_project.language,
+                version: task_runner.library_version(library_name),
+              }
 
-            library = {
-              language: script_project.language,
-              version: task_runner.library_version(library_name),
-            }
+              ProjectDependencies.install(ctx: ctx, task_runner: task_runner)
+              BuildScript.call(ctx: ctx, task_runner: task_runner, script_project: script_project, library: library)
+            end
 
-            ProjectDependencies.install(ctx: ctx, task_runner: task_runner)
-            BuildScript.call(ctx: ctx, task_runner: task_runner, script_project: script_project, library: library)
+            compiled_type = task_runner.compiled_type
```

Comment:
For TS, AS, and Wasm, this will always return `wasm`.
Curious to hear if there's a reason we don't just assume `wasm` here, as opposed to asking the taskrunner for it?

Reply:
I wonder if this is an artifact from before when we had the ability to have non-wasm runtimes (e.g. GraalVM). @jacobsteves do you know what the intention was for this?